amideploy/infrastructure/main.tf

```diff
@@ -0,0 +1,159 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+    tls = {
+      source  = "hashicorp/tls"
+      version = "~> 4.0"
+    }
+  }
+
+  required_version = ">= 1.4.0"
+}
+
+provider "aws" {
+  region  = var.aws_region
+  profile = var.aws_profile
+}
+
+provider "tls" {}
+
+resource "tls_private_key" "airflow_server_private_key" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "generated_airflow_server_key" {
+  key_name   = "auto-generated-key"
+  public_key = tls_private_key.airflow_server_private_key.public_key_openssh
+}
+
+output "airflow_server_private_key_pem" {
+  value     = tls_private_key.airflow_server_private_key.private_key_pem
+  sensitive = true
+}
+
+resource "aws_instance" "ami_connect_airflow_server" {
```

Comment:
One t3-medium EC2 for the webserver and for our compute. (Obviously can adjust the size when we want to).